applications/asset_tracker/src/motion/motion.c

```diff
@@ -109,7 +111,7 @@ static void motion_work_q_handler(struct k_work *work)
 {
 	motion_data_t motion_data;
 
-	if (accelerometer_poll(&motion_data.acceleration) == 0) {
+	if (accelerometer_poll(&motion_data) == 0) {
 		if (get_orientation(&motion_data.orientation,
 					&motion_data.acceleration) == 0) {
 			handler(motion_data);
```

Comment:
You could simply update the timestamp before calling the `handler`, w/o changes in the existing code.

Reply:
Yes, as the library is right now. This was to make room for a potential future expansion where the polling of accelerometer data and retrieving the accelerometer data does not occur at the same time. The most important thing here is that the accelerometer data actually gets timestamped when it is sampled.